File: ProgSem10/primos/primos.py

def eratosthenes2(n):
    multiples = set()
    for i in range(2, n+1):
        if i not in multiples:
            yield i
            multiples.update(range(i*i, n+1, i))

if __name__=="__main__":
    print("Programa que calcula la Sierva de Erathostanes \n")
    n = input("Ingrese el número hasta el cual desea calcular los primos: ")
    print(list(eratosthenes2(int(n))))

File: ProgSem10/primos/primos.pl
%--------------------------------------------------------------
%primos.pl
%Tarea: Jorge David Elihu Mejia Castro
%Predicados principales: allPrimes(Numero primo, P)
%--------------------------------------------------------------

mainPrimos :-
    write("Ingrese hasta que número desea conocer sus primos: "),
    read(P), allPrimes(P,L), print(L).

print([]) :- !.
print(L) :-
    L = [H|T], write(H),write("\n"),
    print(T).

allPrimes(N, Primes) :-
    numlist(2, N, Numlist),
    Stop is round(sqrt(N)),
    allPrimes(Numlist, Stop, Primes).

allPrimes([N|Numlist], Stop, [N|Primes]):-
    exclude(is_multiple(N), Numlist, MPrimes),
    (N =< Stop -> allPrimes(MPrimes, Stop, Primes) ; Primes=MPrimes).

is_multiple(N, I):-
    I mod N =:= 0.
